src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
Let's use here one arrow function for more readability, like:
```suggestion
  const children = people.filter((child) => {
      const hasMother = people.some(person => person.name === child.mother);
      
      return onlyWithSon ? hasMother && child.sex === 'm' : hasMother;
    });
```